### PR TITLE
A couple of in-house PHP conventions

### PR DESCRIPTION
**Pull request by shanebonham**
Here are a couple of PHP conventions for in-house style consideration.

Feedback welcome.

### CODE REVIEW

File: php.md [hunk 1]
```diff
@@ -23,3 +23,33 @@ PHP
     *   [PHPCDC](https://github.com/sebastianbergmann/phpdcd)
     *   [SensioLabs Security Checker](https://github.com/sensiolabs/security-checker)
     *   [PHP Mess Detector](http://phpmd.org)
+
+In-house conventions
```
Comment (jstayton) [inline php.md]: I think we should use title case for headers. I would also call it something like "Coding Style Guide" to be more specific, and mention that it's in addition to PSR-2.

File: php.md [hunk 2]
```diff
@@ -23,3 +23,33 @@ PHP
...
+```html
+<!-- good -->
+<p><?= $var ?></p>
+
+<!-- bad -->
+<p><?=$var?></p>
+<p><?= $var; ?></p>
+```
+
+---
```
Comment (jstayton) [inline php.md]: The convention has been to use number lists for easy reference, rather than separating like this.

**Comment by jstayton** [comment]
I'm good with these additions.

@mcordell @rickyatmonk @kennykaye All in favor?

**Comment by kennykaye** [comment]
:+1: looks good to me!

**Comment by rickyatmonk** [comment]
Looks good. I have a tendency to do `<?= $var; ?>` but it makes sense your way! :+1:

**Comment by shanebonham** [comment]
> Looks good. I have a tendency to do <?= $var; ?> but it makes sense your way! :+1:

Anybody else have any strong feelings about that? I can change, man!

**Comment by shanebonham** [comment]
For what it's worth, [PHP's own docs](http://php.net/manual/en/function.echo.php) do it like this, which is gross:

```php
I have <?=$foo?> foo.
```

**Comment by jstayton** [comment]
I prefer `<?= $foo ?>`.

**Comment by kennykaye** [comment]
I have a tendency to write `<?= $foo; ?>`, but I'm not married to the semicolon by any means.

**Comment by jstayton** [comment]
@shanebonham I think we're good here.

**Comment by mcordell** [comment]
Just a note on the operator alignment, a side effect of this, and possibly why it is not defined, is this problem:

[Consider this block of code](https://github.com/MonkDev/mcms/blob/small-groups/model/small_group_mapper.php#L405-L416)
and [then this block of code](https://github.com/MonkDev/mcms/blob/small-groups/model/small_group_mapper.php#L319-L329). 

The former breaks indentation rules (run it through phpcs) but looks better.

**Comment by shanebonham** [comment]
@mcordell Good point. I think it might call for a clarification in the convention. I would tend to prefer the second, because I think once the alignment includes whitespace and closing parens, the clarity suffers.

I'll write an update.

**Comment by jstayton** [comment]
What about something like this?

```php
$QueryBuilder
    ->leftJoin(
        SmallGroupMemberMapper::DATABASE . '.' . SmallGroupMemberMapper::TABLE,
        self::ID . '=' . SmallGroupMemberMapper::SMALL_GROUP_ID . ' AND ' .
        SmallGroupMemberMapper::ROLE . '=' . SmallGroupMemberMapper::ROLE_PRIMARY_ADMIN
    )
    ->leftJoin(
        UserMapper::DATABASE . '.' . UserMapper::TABLE,
        SmallGroupMemberMapper::USER_ID . '=' . UserMapper::ID
    )
    ->leftJoin(
        SharedUserMapper::DATABASE . '.' . SharedUserMapper::TABLE,
        SharedUserMapper::ID . '=' . UserMapper::ID
    );
```

When something's that gnarly, I tend to do something like that, particularly to avoid lines being too long.

**Comment by shanebonham** [comment]
I can dig it. Does phpcs complain about that?

**Comment by mcordell** [comment]
It does not. I think thats a good solution

**Comment by shanebonham** [comment]
I just pushed an update that shows another `good` example.

**Comment by jstayton** [comment]
Looks good.

**Pull request merged**